app/src/main/java/com/sci/recipeandroid/feature/auth/ui/screen/authentication/LoginFragment.kt

```diff
@@ -0,0 +1,129 @@
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+            loadingIndicator.visibility = View.GONE
+            btnLogin.isEnabled = false
+            btnLogin.setBackgroundColor(
+                ContextCompat.getColor(
+                    requireContext(),
+                    R.color.color_button_inactive
+                )
+            )
+
+            textChangeEventListener()
+
+            //region observe login form state
+            loginViewModel.loginFormState.observe(viewLifecycleOwner) {
+                textInputLayoutEmail.error = it.emailError
+                if (it.email.isNotEmpty() && it.password.isNotEmpty()) {
+                    binding.btnLogin.isEnabled = true
+                    binding.btnLogin.setBackgroundColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.color_border_active
+                        )
+                    )
+                } else {
+                    btnLogin.isEnabled = false
+                    btnLogin.setBackgroundColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.color_button_inactive
+                        )
+                    )
+                }
+            }
+            //endregion
+
+            //region observe login screen event
+            loginViewModel.loginScreenEvent.observe(viewLifecycleOwner){
+                when(it){
+                    is LoginScreenEvent.Error -> {
+                        loadingIndicator.visibility = View.GONE
+                        Snackbar.make(
+                            view,
+                            "Something went wrong", Snackbar.LENGTH_SHORT
+                        ).show()
+                    }
+                    is LoginScreenEvent.Loading -> {
+                        loadingIndicator.visibility = View.VISIBLE
+                        btnLogin.isEnabled = false
+                    }
+                    is LoginScreenEvent.Success -> {
+                        loadingIndicator.visibility = View.GONE
+                        Snackbar.make(
+                            view,
+                            "Login Successful", Snackbar.LENGTH_SHORT
+                        ).show()
+                        requireActivity().supportFragmentManager
+                            .beginTransaction()
+                            .addToBackStack("login")
+                            .replace(R.id.host_fragment, AuthOptionFragment())
+                            .commit()
+                    }
+                }
+            }
+            //endregion
+
+            btnLogin.setOnClickListener {
+                loginViewModel.onEvent(LoginFormEvent.Submit)
+            }
+
+        }
+
+
+    }
+
+    private fun textChangeEventListener() {
```

Comment:
plz "setUpTextChangeEventListener()"

Reply:
changed the name of this function at 9baf80468f57d78d6c4b13f545e1ebaa10f9d59c